app/src/androidTest/java/com/github/siela1915/bootcamp/MainHomeActivityTest.java

```diff
@@ -315,7 +313,71 @@ public void test1(){
         }
     }
 
+    @Test
+    public void isCorrectListOfRecipesDisplayed(){
+        scenario= ActivityScenario.launch(MainHomeActivity.class);
+        //Intents.release();
+
+        Intents.init();
+        onView(withId(R.id.drawer_layout)).perform(DrawerActions.open());
+        onView(withId(R.id.navView)).perform(navigateTo(R.id.menuItem_home));
+        onView(withId(R.id.homeFragment)).check(matches(isDisplayed()));
+        onView(withId(R.id.rand_recipe_recyclerView)).check(matches(isDisplayed()));
+        //onView(withId(R.id.rand_recipe_recyclerView)).check(new RecyclerViewItemCountAssertion(12));
+        onView(withId(R.id.rand_recipe_recyclerView)).perform(actionOnItemAtPosition(0, click()));
+        Intents.intended(hasComponent(RecipeActivity.class.getName()));
+        Intents.release();
+        scenario.close();
 
+    }
+    @Test
+    public void searchViewTest(){
+        scenario= ActivityScenario.launch(MainHomeActivity.class);
+        onView(withId(R.id.searchView)).check(matches(isDisplayed()));
+        onView(withId(R.id.searchView)).perform(click());
+        onView(withId(R.id.searchView)).perform(ViewActions.typeText("pizza"));
+        ViewActions.closeSoftKeyboard();
+        onView(withText("pizza")).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+        scenario.close();
+    }
+    @Test
+    public void moreFiltersTest(){
+        scenario= ActivityScenario.launch(MainHomeActivity.class);
+        onView(withId(R.id.moreFilterTextView)).check(matches(isDisplayed()));
+        onView(withId(R.id.moreFilterTextView)).perform(click());
+        onView(withId(R.id.filter)).check(matches(isDisplayed()));
+        //onView(withId(R.id.btnPrpTime)).perform(click());
+        //onView(withText("Choose the preparation time")).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+        onView(withId(R.id.btnDiet)).perform(click());
+        onView(withText("Choose your diet")).check(matches(withEffectiveVisibility(VISIBLE)));
+        scenario.close();
+    }
+    @Test
+    public void moreFilterByClickingOonAllergyBtnTest(){
+        scenario= ActivityScenario.launch(MainHomeActivity.class);
+        onView(withId(R.id.moreFilterTextView)).perform(click());
+        onView(withId(R.id.btnAllergy)).perform(click());
+        onView(withText("what are you allergic to")).check(matches(withEffectiveVisibility(VISIBLE)));
+        scenario.close();

```

Comment:
To keep the test code clean, you might consider separating setup and teardown code by moving them to a separate `@Before` and an `@After` method instead of launching the activity in each test method.